lldb/source/Utility/Log.cpp

```diff
@@ -438,3 +439,15 @@ void RotatingLogHandler::Dump(llvm::raw_ostream &stream) const {
   }
   stream.flush();
 }
+
+TeeLogHandler::TeeLogHandler(std::shared_ptr<LogHandler> first_log_handler,
+                             std::shared_ptr<LogHandler> second_log_handler)
+    : m_first_log_handler(first_log_handler),
+      m_second_log_handler(second_log_handler) {}
+
+void TeeLogHandler::Emit(llvm::StringRef message) {
+  if (m_first_log_handler)
```

Comment:
Is the idea that one will ne optional in the typical use-case?

Reply:
No, it's just being resilient agains the shared_ptrs being null. The alternative would be an assert in the ctor. Let me know if you prefer that? It would be slightly more efficient.